js/sales.js

```diff
@@ -24,56 +24,119 @@
+                newHourHead.textContent = (militaryHour - standardTimeOffSet) + ':00pm';
+            }
+            tableRow.appendChild(newHourHead);
+        }
+        return tableRow;
+    }
+
+    CookieStore.prototype.render = function(){
+        var tableStoreRow = document.createElement('tr');
+        tableStoreRow.setAttribute('scope', 'row');
+        var tableStoreHead = document.createElement('th');
+        tableStoreHead.textContent = this.name;
+        document.querySelector('tbody').appendChild(tableStoreRow);
+        tableStoreRow.appendChild(tableStoreHead);
+
+        calculateSimulatedCookiesPerHour(this);
+        
+        for(var i = 0; i < this.simulatedCookiesPerHour.length; i++){
+            var tableStoreData = document.createElement('td');
+            tableStoreData.textContent = this.simulatedCookiesPerHour[i];
+            tableStoreRow.appendChild(tableStoreData);
+        }
+        return tableStoreRow;
+    }
+
+    function createTotalsFooterRow(){
+        var totalsRow = document.createElement('tr');
+        totalsRow.setAttribute('scope', 'row');
+        var totalsHead = document.createElement('th');
+        totalsHead.textContent = 'Totals';
+        document.querySelector('tfoot').appendChild(totalsRow);
+        totalsRow.appendChild(totalsHead);
+
+        for(var i = 0; i < cookieStoreLocations[0].simulatedCookiesPerHour.length; i++){
+            var total = 0;
+            for(var j = 0; j < cookieStoreLocations.length; j++){
+                total += cookieStoreLocations[j].simulatedCookiesPerHour[i];
             }
-            list.append(newHourItem);
+            var tableTotalData = document.createElement('td');
+            tableTotalData.textContent = total;
+            totalsRow.appendChild(tableTotalData);
+        }
+        return totalsRow;
+    }
+
+    function createStoreTotals(location, tableRow){
+        var total = 0;
+        for(var i = 0; i < location.simulatedCookiesPerHour.length; i++){
+            total += location.simulatedCookiesPerHour[i];
         }
+        var tableStoreTotalData = document.createElement('td');
+        tableStoreTotalData.textContent = total;
+        tableRow.appendChild(tableStoreTotalData);
+        return total;
+    }
+
+    function createDailyLocationTotalHeader(tableRow){
+        var dailyLocationTotal = document.createElement('th');
+        dailyLocationTotal.setAttribute('scope', 'col');
+        dailyLocationTotal.textContent = 'Daily Location Total';
+        tableRow.appendChild(dailyLocationTotal);
     }
 
-    createStoreLocationHTMLSections(cookieStoreLocations);
- 
+    function totalOfAllStoresDaily(tableRow, allStoresTotal){
+        var totalOfAllStoresDailyTableData = document.createElement('td');
+        totalOfAllStoresDailyTableData.textContent = allStoresTotal;
+        tableRow.appendChild(totalOfAllStoresDailyTableData);
+    }
```

Comment:
I think we should revisit all these functions that mutate a `<tr>` element and have them return a value that is then appended to the element, instead. It would remove a parameter and create code that relies on `return` values (referred to as [functional programming](https://en.wikipedia.org/wiki/Functional_programming)). Functional programming is generally easier to read, maintain, reuse, and scale.